src/ds/thread_messaging.h

```diff
@@ -65,6 +65,41 @@ namespace threading
 #endif
     }
 
+    ~Task()
+    {
+#ifdef USE_MPSCQ
+      while (!queue.is_emtpy())
```

Comment:
```suggestion
      while (!queue.is_empty())
```

Worth building this with `USE_MPSCQ` to confirm it still compiles.

Reply:
Yes, I'll have a look.